demonstrations/tutorial_pulse_programming101.py

```diff
@@ -0,0 +1,362 @@
+    :property="og:image": https://pennylane.ai/qml/_images/pauli_shadows.jpg
+
+*Author: Korbinian Kottmann — Posted: 20 February 2023.
+
+In this demo we are going to introduce pulse gates and differentiable pulse programming, showcase
+the current functionality in PennyLane and run the ctrl-VQE algorithm for an example molecule.
+
+Pulses in quantum computers
+---------------------------
+
+In non-measurement-based quantum computers such as superconducting and ion trap systems, qubits are realized through physical systems with a discrete set of energy levels.
+For example, transmon qubits realize an anharmonic oscillator whose ground and first excited states can serve as the two energy
+levels of a qubit. Such a qubit can be controlled via an electromagnetic field tuned to its energy gap. In general, this
+electromangnetic field can be altered in time, leading to a time-dependent Hamiltonian interaction :math:`H(t)`.
+We call driving the system with such an electromagnetic field for a fixed time window a pulse sequence. During a pulse sequence, the state evolves according
+to the time-dependent Schrodinger equation
+
+.. math:: \frac{d}{dt}|\psi\rangle = -i H(t) |\psi\rangle
+
+realizing a unitary evolution :math:`U(t_0, t_1)` from times :math:`t_0` to :math:`t_1` of the input state, i.e. 
+:math:`|\psi(t_1)\rangle = U(t_0, t_1) |\psi(t_0)\rangle`.
+
+In non-measurement-based digital quantum computers, the amplitude and frequencies of predefined pulse sequences are
+fine tuned to realize the native gates of the quantum computer. More specifically, the Hamiltonian interaction :math:`H(t)`
+is tuned such that the respective evolution :math:`U(t_0, t_1)` realizes for example a Pauli or CNOT gate.
+
+Pulse programming in PennyLane
+------------------------------
+
+A user of a quantum computer typically operates on the higher and more abstract gate level.
+Future fault tolerance quantum computers require this abstraction to allow for error correction.
+For noisy and intermediate sized quantum computers, the abstraction of decomposing quantum algorithms
+into a fixed native gate set can be a hindrance and unnecessarily increase execution time, therefore leading
+to more decoherence. The idea of differentiable pulse programming is to optimize quantum circuits on the pulse
+level with the aim of achieving the shortest interaction sequence a hardware system allows.
+
+In PennyLane, we can now simulate arbitrary qubit system interactions to explore the possibilities of such pulse programs.
+First, we need to define the time-dependent Hamiltonian :math:`H(p, t)= \sum_i f_i(p, t) H_i` with envelope :math:`f_i(p, t)` that may
+depend on parameters :math:`p` and constant operators :math:`H_i`. In PennyLane, we can do this intuitively in the following way.
+"""
+
+import pennylane as qml
+import pennylane.numpy as np
+import jax.numpy as jnp
+import jax
+import matplotlib.pyplot as plt
+
+def f1(p, t):
+    return jnp.polyval(p, t)
```

Comment:
Do we want this, maybe?
```suggestion
f1 = jnp.polyval
```

Reply:
omg I never thought of this 😅 🤯 🙈

Reply:
Does this not work? :)

Reply:
It does, I just found this more self-explanatory seeing the signature. I'd be slightly confused seeing just `f1 = jnp.polyval` the first time